cmake/extensions.cmake

```diff
@@ -220,11 +238,29 @@ Please provide one of following: CONF_FILES")
     set(${ZEPHYR_FILE_DTS} ${${ZEPHYR_FILE_DTS}} PARENT_SCOPE)
   endif()
 
-  if(NOT DEFINED ZEPHYR_FILE_BOARD)
+  if(NOT DEFINED additional_append)
     # Defaulting to system wide settings when BOARD is not given as argument
-    set(ZEPHYR_FILE_BOARD ${BOARD}${BOARD_QUALIFIERS})
+    set(board_in ${BOARD})
+
     if(DEFINED BOARD_REVISION)
-      set(ZEPHYR_FILE_BOARD_REVISION ${BOARD_REVISION})
+      set(board_in ${board_in}@${BOARD_REVISION})
+    endif()
+
+    set(board_in ${board_in}${BOARD_QUALIFIERS})
+    parse_board_components(board_in board_name board_revision board_qualifiers)
```

Comment:
first you concatenate board, qualifiers, revision together, just to afterwards split them again, why ?

Reply:
because zephyr_file calls zephyr_build_string which cannot handle all of these being provided in a single BOARD argument, they must be separated